spec/services/get_product_service_spec.rb

```diff
@@ -39,15 +47,34 @@
   end
 
   describe '#call' do
-    context 'when validation succeeds' do
-      let(:product_id) { tea.id }
+    include_context 'redis mock'
+    let(:product_id) { tea.id }
+
+    context 'when a product exists in cache' do
+      before do
+        allow(CacheManager).to receive(:fetch_value).and_return(tea.to_json)
+      end
+      it 'return cached product' do
+        service.call
+
+        expect(service.success?).to eq true
+        expect(service.result['id']).to eq tea.id
+        expect(service.result['id']).not_to eq milk_tea.id
```

Comment:
only 1 product, if the id is equal to tea, it is obvious not eq to milk_tea => unnecessary expect